RFS/src/testFixtures/java/com/rfs/framework/PreloadedDataContainerOrchestrator.java

```diff
@@ -0,0 +1,285 @@
+    public static String PRELOADED_IMAGE_BASE_NAME = "migrations/preloaded_source_";
+    public static int PULL_TIMEOUT_SECONDS = 600;
+
+    @EqualsAndHashCode
+    @AllArgsConstructor
+    private static class HashHelper {
+        String baseSourceImageId;
+        String dataLoaderImageId;
+        String[] dataLoaderArgs;
+    }
+
+    private final ElasticsearchContainer.Version baseSourceVersion;
+    private final String serverNameAlias;
+    private final String dataLoaderImageName;
+    private final String[] generatorContainerArgs;
+
+
+    public PreloadedDataContainerOrchestrator(ElasticsearchContainer.Version baseSourceVersion,
+                                              String serverNameAlias,
+                                              String dataLoaderImageName,
+                                              String[] generatorContainerArgs)
+    {
+        this.baseSourceVersion = baseSourceVersion;
+        this.serverNameAlias = serverNameAlias;
+        this.dataLoaderImageName = dataLoaderImageName;
+        this.generatorContainerArgs = generatorContainerArgs;
+    }
+
+    public String getReadyImageName(boolean pullIfUnavailable) throws IOException, InterruptedException {
+        var imageName = getImageName();
+        var dockerClient = createDockerClient();
+        var tag = Integer.toString(Math.abs(getHashCodeOfImagesAndArgs(dockerClient, pullIfUnavailable)));
+        if (getExistingImage(dockerClient, imageName, tag, false) == null) {
+            makeNewImage(dockerClient, imageName, tag);
+        }
+        return formatFullImageName(imageName, tag);
+    }
+
+    String[] getImageAndTagArray(String imageName) {
+        var imageTagArray = imageName.split(":");
+        if (imageTagArray.length != 2) {
+            throw new IllegalArgumentException("Base source image [" +  baseSourceVersion.imageName +
+                    "] name isn't of the form .*:.*");
+        }
+        return imageTagArray;
+    }
+
+    String getImageId(DockerClient dockerClient, String imageName, boolean pullIfUnavailable)
+            throws InterruptedException
+    {
+        var imageAndTagArr = getImageAndTagArray(imageName);
+        var image = getExistingImage(dockerClient, imageAndTagArr[0], imageAndTagArr[1], pullIfUnavailable);
+        if (image == null) {
+            throw new IllegalStateException("Base source image doesn't exist [" + baseSourceVersion.imageName +
+                    "].  Please build/pull it first.");
+        }
+        return image.getId();
+    }
+
+    private int getHashCodeOfImagesAndArgs(DockerClient dockerClient, boolean pullIfUnavailable)
+        throws InterruptedException
+    {
+        return new HashHelper(
+                getImageId(dockerClient, baseSourceVersion.imageName, pullIfUnavailable),
+                getImageId(dockerClient, dataLoaderImageName, pullIfUnavailable),
+                generatorContainerArgs).hashCode();
```

Comment:
Pretty sure you can remove HashHelper entirely
```suggestion
        return Objects.hash(
                getImageId(dockerClient, baseSourceVersion.imageName, pullIfUnavailable),
                getImageId(dockerClient, dataLoaderImageName, pullIfUnavailable),
                generatorContainerArgs);
```